site/content/en/docs/manual/advanced/ai-tools.md

```diff
@@ -1,155 +1,309 @@
+by placing points with the automatic drawing of a line between them.
+The distance between the adjacent points is limited by the threshold of action,
+displayed as a red square that is tied to the cursor.
+
+To use **Intelligent scissors**, do the following:
+
+1. In the **OpenCV** menu, go to the **Drawing** tab.
+2. Select the label, and then click on the **Intelligent scissors** button.
+
+![](/images/image199.jpg)
+
+3. Add the first point on the boundary of the allocated object. <br> You will see a line repeating the outline of the object.
+4. Add the second point, so that the previous point is within the restrictive threshold.
+   <br>After that a line repeating the object boundary will be automatically created between the points.
+   ![](/images/image200_detrac.jpg)
+5. To finish placing points, on the top menu**Done** (or **N** on the keyboard).
 
-Fast online Object Tracking and Segmentation. Tracker is able to track different objects in one server request.
-Trackable object will be tracked automatically if the previous frame was
-a latest keyframe for the object. Have tracker indication on canvas. `SiamMask` tracker supported CUDA.
+As a result, a polygon will be created
+For more information, see [Annotation with polygons](/docs/manual/advanced/annotation-with-polygons/).
 
-> If you plan to track simple non-overlapping objects consider using fast client-side [TrackerMIL from OpenCV](/docs/manual/advanced/opencv-tools/#trackermil).
+To increase or lower the action threshold, hold **Ctrl** and scroll the mouse wheel.
+Increasing the action threshold will affect the performance.
+During the drawing process, you can remove the last point by clicking on it with the left mouse button.
+
+When tracking:
+
+- You can create a boundary manually (like when
+  [creating a polygon](/docs/manual/advanced/annotation-with-polygons/manual-drawing/))
+  by temporarily disabling the automatic line creation. To do that, click `Ctrl`.
+
+- You can change the number of points in the polygon with slider:
+
+  ![](/images/image224.jpg)
+
+- To change polygon opacity, on the [Objects sidebar](/docs/manual/basics/objects-sidebar/#appearance)
+  use the **Selected opacity** slider.
+
+### Histogram Equalization
+
+Histogram equalization is a CV method that improves contrast in an
+image to stretch out the intensity range.
+This method usually increases the global contrast of images when its usable data
+is represented by close contrast values.
+It is useful in images with backgrounds and foregrounds that are both bright or dark.
+
+1. In the **OpenCV** menu, go to the **Image** tab.
+2. Click on **Histogram equalization** button.
+   <br>![](/images/image221.jpg)
+
+Then contrast of the current frame will be improved.
+If you change the frame, it will be equalized too.
+You can disable equalization by clicking the **Histogram equalization** button again.
+
+![](/images/image222.jpg)
+
+### TrackerMIL
```

Comment:
I don't understand why we have here information about MIL tracker. It should be in the tracker section only.